File: lab6/computercomponents/urls.py
from django.contrib import admin
from computercomponents import views as comp_views
from django.urls import include, path
from rest_framework import routers

router = routers.DefaultRouter()
router.register(r'cpu', comp_views.CpuViewSet)
router.register(r'disk', comp_views.DiskstorageViewSet)
router.register(r'ram', comp_views.RamViewSet)
router.register(r'hardware', comp_views.HardwareViewSet)

urlpatterns = [
    path('api/', include(router.urls)),
    path('', comp_views.index),
    path('<str:model_name>/', comp_views.list, name='list')
]

File: dz/src/modules/urls.js
class Urls {
    constructor() {
        this.url = 'http://localhost:8000/api/';
    }

    cpus() {
        return `${this.url}cpu/`
    }

    cpu(id) {
        return `${this.url}cpu/${id}/`
    }

    disks() {
        return `${this.url}disk/`
    }

    disk(id) {
        return `${this.url}disk/${id}/`
    }

    rams() {
        return `${this.url}ram/`
    }

    ram(id) {
        return `${this.url}ram/${id}/`
    }

    hardwares() {
        return `${this.url}hardware/`
    }

    hardware(id) {
        return `${this.url}hardware/${id}/`
    }
}

export const urls = new Urls()